packages/contracts-bedrock/src/L2/RevenueSharer.sol

```diff
@@ -0,0 +1,175 @@
+     */
+    address payable public BENEFICIARY;
+    /**
+     * @dev The address of the L1 wallet that will receive the OP chain runner's share of fees.
+     */
+    address public L1_WALLET;
+
+    /*//////////////////////////////////////////////////////////////
+                            Events
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev Emitted when fees are disbursed.
+     * @param _share The amount of fees shared to the Beneficiary.
+     * @param _total The total funds distributed.
+     */
+    event FeesDisbursed(uint256 _share, uint256 _total);
+    /**
+     * @dev Emitted when fees are received from FeeVaults.
+     * @param _sender The FeeVault that sent the fees.
+     * @param _amount The amount of fees received.
+     */
+    event FeesReceived(address indexed _sender, uint256 _amount);
+
+    /*//////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev Constructor for the FeeDisburser contract which validates and sets immutable variables.
+     * @param _beneficiary The address which receives the revenue share.
+     * @param _l1Wallet The L1 address which receives the remainder of the revenue.
+     */
+    constructor(address payable _beneficiary, address payable _l1Wallet) {
+        initialize(_beneficiary, _l1Wallet);
+    }
+
+    function initialize(address payable _beneficiary, address payable _l1Wallet) public initializer {
+        if (_beneficiary == address(0)) revert ZeroAddress("_beneficiary");
+        if (_l1Wallet == address(0)) revert ZeroAddress("_l1Wallet");
+        BENEFICIARY = _beneficiary;
+        L1_WALLET = _l1Wallet;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            External Functions
+    //////////////////////////////////////////////////////////////*/
+    /**
+     * @dev Withdraws funds from FeeVaults, sends Optimism their revenue share, and withdraws remaining funds to L1.
+     */
+    function execute() external virtual {
+        // Pull in revenue
+        uint256 d = feeVaultWithdrawal(Predeploys.L1_FEE_VAULT);
+        uint256 b = feeVaultWithdrawal(Predeploys.BASE_FEE_VAULT);
+        uint256 q = feeVaultWithdrawal(Predeploys.SEQUENCER_FEE_WALLET);
+
+        // Compute expenditure
+        uint256 e = getL1FeeExpenditure();
+
+        // Compute revenue and profit
+        uint256 r = d + b + q; // revenue
+        uint256 p = r - e; // profit
+
+        // Compute revenue share
+        uint256 s = Math.max(
+            REVENUE_COEFFICIENT_BASIS_POINTS * r / BASIS_POINT_SCALE,
+            PROFIT_COEFFICIENT_BASIS_POINTS * p / BASIS_POINT_SCALE
+        ); // share
+        uint256 remainder = r - s;
+
+        // Send Beneficiary their revenue share on L2
+        if (!SafeCall.send(BENEFICIARY, gasleft(), s)) {
```

Comment:
i would like us to add a version of `SafeCall` that only takes 2 arguments so we don't need to pass `gasleft`. If you don't want to do it as part of this PR thats fine